src/node.rs

```diff
@@ -545,6 +545,25 @@ impl<'a> NodeProperty<'a> {
         }
     }
 
+    /// Attempts to parse the property value as a list of [`usize`].
+    pub fn as_usize_list(self, cell_size: usize) -> impl Iterator<Item = usize> + 'a {
+        let chunk_len = match cell_size {
+            1 => 4,
+            2 => 8,
+            _ => 1,
+        };
```

Comment:
Some input validation (and returning `Option<...>` or `Result<..., ...>`) instead of defaulting to `1` would be preferred, or in the worst case publicly documenting the set of accepted `cell_size` values that will produce expected outcomes. Also probably worth noting explicitly that it won't handle property values that have mixed cell size values and that for platforms where `size_of::<u64>() > size_of::<usize>()` will truncate the values. (so potentially returning `u64`s instead of `usize`s might be preferable)

Reply:
> Some input validation (and returning Option<...> or Result<..., ...>) instead of defaulting to 1 would be preferred

I was actually thinking of adding a `enum CellSize` type with valid variants, something like:

```rust
pub enum CellSize {
    One = 1,
    Two = 2,
    None = 0,
}
```

The input validation is not explicitly necessary, since `4` and `8` are the only valid values here. The user input `1`, `2` are also validated by that match statement, since they are the only variants that produce valid outputs.

However, if another type would be preferred to clarify intent for people reading the code, maybe another `enum` would be useful:

```rust
pub enum IntValueLen {
    Four = 4,
    Eight = 8,
    None = 0,
}
```

>  Also probably worth noting explicitly that it won't handle property values that have mixed cell size values

See my comment about adding a `pub fn iter_addr_size_cell_list` for another function that handles the difference in a `prop-encoded-array`. Are there other primitives in the Device Tree spec that have differing cell sizes?

> so potentially returning u64s instead of usizes might be preferable

That sounds like it might be worth doing, but would also imply every other location that returns a `usize` value should be changed to a `u64` (or corresponding expected fixed-size type). Maybe that should be saved for a separate PR?